### DISCUSSION TITLE
Pyinstaller with "--onefile" tag does not add the data for button icons

### DISCUSSION THREAD

**Discussion by Ernesto-Alves67** [pyinstaller/pyinstaller#8563] [PyInstaller] [answered]
Recently i made a program that render markdown in a desktop aplicattion without internet.

## So the problem is: If i run this

```
@echo off
echo Starting Build mk.py with PyInstaller...
pyinstaller --noconsole --name DocViewer --icon="doc_icon.ico" --add-data "ui_docV.py;." --add-data "icons_svg/*;icons"  mk.py
echo Build complete. Press any key to exit.

```
## the icons are correctly added to de final exe.
<img src="https://github.com/ScherzoLambda/DocViewer_Palace/assets/98620218/7045e975-808a-4d39-877a-2b2a829fc505" alt="docViewer" width="500" >

## But if i run this
```
pyinstaller --onefile --noconsole --name DocViewer --icon="doc_icon.ico" --add-data "ui_docV.py;." --add-data "icons_svg/*;icons" mk.py
```
## the icons doesnt appear in the final exe.
<img src="https://github.com/pyinstaller/pyinstaller/assets/98620218/1ebd3fcd-4c26-4429-9a0d-c54dee89f99b" width="500">

## Why this happens? Also looking for a way to reduce de size of my program. Can you guys help me?

**Repo Link:** https://github.com/ScherzoLambda/DocViewer_Palace/tree/main/MarkViewDesktop

**Comment by bwoodsend** [comment]
You're probably referencing the icons in your Python code using relative paths which is invalid. See [docs](https://pyinstaller.org/en/stable/runtime-information.html#using-file).

**Reply by Ernesto-Alves67** [reply]
I'm doing like this 

```
class Ui_MainWindow(object):
    iconspath = "_internal\\"+ "\\icons" + "\\"

.....
......

   self.heading_btn.setIcon(QIcon(loadSvgIcon(self.iconspath+"bx-heading.svg"))
```

**Reply by bwoodsend** [reply]
Yep, that's invalid. Use `__file__` like the docs say.

**Reply by Ernesto-Alves67** [reply, marked as answer]
I didn't understand this `__file__` aproach, i find a diferent one that works for me.

```Python
def resource_path(relative_path):
    """ Get the absolute path to the resource, works for dev and for PyInstaller """
    try:
        base_path = sys._MEIPASS
    except AttributeError:
        base_path = path.abspath(".")

    return path.join(base_path, relative_path)
    
......
.......
class Ui_MainWindow(object):
         ...
        def setupUi(self, MainWindow):
               self.heading_btn.setIcon(QIcon(loadSvgIcon(resource_path('icons/bx-heading.svg'))))
```

## Thanks for help me.

**Reply by bwoodsend** [reply]
That's invalid too. It'll break under normal Python if ran from a different working directory. Try:  `os.path.join(os.path.dirname(__file__), "icons/bx-heading.svg")`
